packages/noya-state/src/selectors/pageSelectors.ts

```diff
@@ -12,7 +13,9 @@ export type PageMetadata = {
   scrollOrigin: Point;
 };
 
-export const getCurrentPageIndex = (state: ApplicationState) => {
+export const getCurrentPageIndex = (
+  state: ApplicationState | Draft<ApplicationState>,
```

Comment:
`getCurrentPageIndex` and `getCurrentPage` can probably just use `Draft<ApplicationState>`